Jenkinsfile.trigger

```
@@ -102,69 +102,98 @@ if (breakEarly) { return } // thanks Jenkins...
+	for (int i = 0; i < queue.size(); i++) {
+		def buildObj = queue[i]
 
 		// "catchError" to set "stageResult" :(
 		catchError(message: 'Build of "' + buildObj.identifier + '" failed', buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
 			if (buildObj.gha_payload) {
-				node {
-					withEnv([
-						'payload=' + buildObj.gha_payload,
-					]) {
-						withCredentials([
-							string(
-								variable: 'GH_TOKEN',
-								credentialsId: 'github-access-token-docker-library-bot-meta',
-							),
+				stage(buildObj.identifier) {
+					node {
+						withEnv([
+							'payload=' + buildObj.gha_payload,
 						]) {
-							sh '''
-								set -u +x
-
-								# https://docs.github.com/en/free-pro-team@latest/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event
-								curl -fL \
-									-X POST \
-									-H 'Accept: application/vnd.github+json' \
-									-H "Authorization: Bearer $GH_TOKEN" \
-									-H 'X-GitHub-Api-Version: 2022-11-28' \
-									https://api.github.com/repos/docker-library/meta/actions/workflows/build.yml/dispatches \
-									-d "$payload"
-							'''
+							withCredentials([
+								string(
+									variable: 'GH_TOKEN',
+									credentialsId: 'github-access-token-docker-library-bot-meta',
+								),
+							]) {
+								sh '''
+									set -u +x
+
+									# https://docs.github.com/en/free-pro-team@latest/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event
+									curl -fL \
+										-X POST \
+										-H 'Accept: application/vnd.github+json' \
+										-H "Authorization: Bearer $GH_TOKEN" \
+										-H 'X-GitHub-Api-Version: 2022-11-28' \
+										https://api.github.com/repos/docker-library/meta/actions/workflows/build.yml/dispatches \
+										-d "$payload"
+								'''
+							}
 						}
+						// record that GHA was triggered (for tracking continued triggers that fail to push an image)
+						buildCompletionData[buildObj.buildId] = [
+							lastTime: System.currentTimeMillis() / 1000, // convert to seconds
+							url: currentBuild.absoluteUrl,
+						]
 					}
-					// record that GHA was triggered (for tracking continued triggers that fail to push an image)
-					buildCompletionData[buildObj.buildId] = [
-						lastTime: System.currentTimeMillis() / 1000, // convert to seconds
-						url: currentBuild.absoluteUrl,
-					]
 				}
 			} else {
+				// why not parallel these build() invocations?
+				// jenkins parallel closures get started in a randomish order, ruining our sorted queue
 				def res = build(
 					job: 'build',
 					parameters: [
+						string(name: 'identifier', value: buildObj.identifier),
 						string(name: 'buildId', value: buildObj.buildId),
 					],
 					propagate: false,
-					quietPeriod: 5, // seconds
+					// trigger these quickly so they all get added to Jenkins queue in "queue" order
+					quietPeriod: 0, // seconds
+					// we'll wait on them after they are all queued
+					waitForStart: true,
```

We can't have a `quietPeriod` on these (even though that would be nice) because it messes with `waitForStart` (and we would then be "captive" for the entire quiet period) -- I think we should probably note that here. :thinking:

Maybe something like this?

```suggestion
					// trigger these quickly so they all get added to Jenkins queue in "queue" order (also using "waitForStart" means we have to wait for the entire "quietPeriod" before we get to move on and schedule more)
					quietPeriod: 0, // seconds
					// we'll wait on the builds in parallel after they are all queued (so our sorted order is the queue order)
					waitForStart: true,
```